datatypes/src/collections/feature_collection_impl.rs

```diff
@@ -172,6 +172,45 @@ pub trait FeatureCollectionImplHelpers {
 
     /// Is the feature collection simple or does it contain multi-features?
     fn _is_simple(&self) -> bool;
+
+    fn f64_michael_equals(a: &Float64Array, b: &Float64Array) -> bool {
```

Comment:
while i appreciate method names indicating their functionality, i suggest to change this to "f64_eq_with_nan_eq_nan". Later generations might not know what a michael is in this context.

Reply:
```suggestion
    /// compares two f64 typed columns
    /// treats NaN values as if they are equal
    fn f64_column_equals(a: &Float64Array, b: &Float64Array) -> bool {
```